libqpdf/QPDFParser.cc

```diff
@@ -364,15 +348,13 @@ QPDFParser::parseRemainder(bool content_stream)
             continue;
 
         case QPDFTokenizer::tt_name:
-            {
-                auto const& name = tokenizer.getValue();
-                addScalar<QPDF_Name>(name);
-
-                if (name == "/Contents") {
-                    b_contents = true;
-                } else {
-                    b_contents = false;
-                }
+            if (frame->state == st_dictionary_key) {
+                frame->key = tokenizer.getValue();
+                frame->state = st_dictionary_value;
+                b_contents = decrypter && frame->key == "/Contents";
```

Comment:
Can you check b_contents carefully. I tried to read the spec but I do not feel confident in my understanding.

Reply:
I went through it carefully. It looks correct.